shell/platform/linux/fl_key_event_plugin.cc

```diff
@@ -20,11 +24,113 @@ static constexpr char kUnicodeScalarValuesKey[] = "unicodeScalarValues";
 static constexpr char kGtkToolkit[] = "gtk";
 static constexpr char kLinuxKeymap[] = "linux";
 
+static constexpr uint64_t kMaxPendingEvents = 1000;
+
+// Declare and define a private pair object to bind the id and the event
+// together.
+
+G_BEGIN_DECLS
+G_DECLARE_FINAL_TYPE(FlKeyEventPair,
+                     fl_key_event_pair,
+                     FL,
+                     KEY_EVENT_PAIR,
+                     GObject);
+G_END_DECLS
+
+struct _FlKeyEventPair {
+  GObject parent_instance;
+
+  uint64_t id;
+  GdkEventKey* event;
+};
+
+G_DEFINE_TYPE(FlKeyEventPair, fl_key_event_pair, G_TYPE_OBJECT)
+
+static void fl_key_event_pair_dispose(GObject* object) {
+  g_return_if_fail(FL_IS_KEY_EVENT_PAIR(object));
+  FlKeyEventPair* self = FL_KEY_EVENT_PAIR(object);
+  gdk_event_free(reinterpret_cast<GdkEvent*>(self->event));
+}
```

Comment:
Missing the call to the parent dispose method:
`G_OBJECT_CLASS(fl_key_event_pair_parent_class)->dispose(object);`